Library/Homebrew/rubocops/cask/url.rb

```diff
@@ -24,19 +24,33 @@ class Url < Base
         def on_url_stanza(stanza)
           return if stanza.stanza_node.block_type?
 
+          url_stanza = stanza.stanza_node.first_argument
           hash_node = stanza.stanza_node.last_argument
           return unless hash_node.hash_type?
 
           hash_node.each_pair do |key_node, value_node|
             next unless key_node.source == "verified"
             next unless value_node.str_type?
-            next unless value_node.source.start_with?(%r{^"https?://})
+
+            if value_node.source.start_with?(%r{^"https?://})
+              add_offense(
+                value_node.source_range,
+                message: "Verified URL parameter value should not contain a URL scheme.",
+              ) do |corrector|
+                corrector.replace(value_node.source_range, value_node.source.gsub(%r{^"https?://}, "\""))
+              end
+            end
+
+            # Skip if the URL and the verified value are the same.
+            next if value_node.str_content == url_stanza.str_content.gsub(%r{^https?://}, "")
```

Comment:
Hmm, this has a problem in real casks compared to my test cases. 😒 `url_stanza = stanza.stanza_node.first_argument.str_content` isn't always the string content because `url`s with interpolation could be `dstr` etc:

```
s(:send, nil, :url,
  s(:dstr,
    s(:str, "https://azuredatastudio-update.azurewebsites.net/"),
    s(:begin,
      s(:send, nil, :version)),
    s(:str, "/"),
    s(:begin,
      s(:send, nil, :arch)),
    s(:str, "/stable")),
  s(:hash,
    s(:pair,
      s(:sym, :verified),
      s(:str, "azuredatastudio-update.azurewebsites.net/"))))
```

Reply:
Fixed.